pkg/block/local/adapter.go

```diff
@@ -367,12 +366,34 @@ func (l *Adapter) GetProperties(_ context.Context, obj block.ObjectPointer) (blo
 		}
 		return block.Properties{}, err
 	}
-	// No properties, just return that it exists
+
+	etag, err := calcFileETag(p)
+	if err != nil {
+		return block.Properties{}, err
+	}
+
 	return block.Properties{
 		LastModified: stat.ModTime(),
+		ETag:         etag,
 	}, nil
 }
 
+func calcFileETag(fileKey string) (string, error) {
```

Comment:
This unnecessary refactoring makes it much more difficult to review the PR

Reply:
You moved this function from walker.go to adapter.go  both are under the local package so I don't understand why it is necessary